mautic/mautic
# GrapesJS RTE disable fix

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [Yes]
| New feature/enhancement? (use the a.x branch)      | [No]
| Deprecations?                          | [No]
| BC breaks? (use the c.x branch)        | [No]
| Automated tests included?              | [No] 
| Related user documentation PR URL      | #
| Related developer documentation PR URL | #
| Issue(s) addressed                     | #

#### Description:

PR fixes a bug that occurs in the GrapesJS MJML email CKEditor. It was reported in GrapesJS MJML library 2 years ago - [grapesjs-mjml-issue-193](https://github.com/artf/grapesjs-mjml/issues/193) but so far has not been finally resolved and merged. The problem occurs when the user tries to style one of the text blocks of MJML email using the CKEditor toolbar and Style Manager. This PR adjusts bug solution prepared in the [commit](https://github.com/yanfanvip/grapesjs-plugin-ckeditor/commit/52123968877a2660265e322b875d13225d00dc94)

#### Steps to test this PR:

1. Open MJML email for editing or open [GrapesJS MJML Demo](https://grapesjs.com/demo-mjml.html)
2. Select one of the text blocks and style the message with text editor toolbar.
3. In Style Manager => Typography, change e.g. font style or text decoration.
4. Select the same text block again and try to style it using the text editor toolbar e.g. bold the text.
5. Toolbar stops responding. Attempting to convert text to a list causes the following error in the browser console:
```
ckeditor.js:186 Uncaught TypeError: Cannot read properties of null (reading 'isContextFor')
at window.CKEDITOR.window.CKEDITOR.dom.CKEDITOR.command.refresh (ckeditor.js:186:481)
at window.CKEDITOR.window.CKEDITOR.dom.CKEDITOR.command.exec (ckeditor.js:715:448)
at window.CKEDITOR.window.CKEDITOR.dom.CKEDITOR.command.exec (ckeditor.js:186:379)
at a.execCommand (ckeditor.js:252:231)
at CKEDITOR.tools.extend.click (ckeditor.js:766:419)
at Object.execute (ckeditor.js:767:490)
at ckeditor.js:768:326
at ckeditor.js:29:180
at Object.callFunction (ckeditor.js:29:296)
at HTMLAnchorElement.onclick (demo-mjml.html:1:16)
```
6. Repeat previous steps with this PR on Gitpod or download for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester)), error should no longer occur.
